src/coreclr/src/vm/gccover.cpp

```diff
@@ -1446,6 +1463,17 @@ void DoGcStress (PCONTEXT regs, NativeCodeVersion nativeCodeVersion)
 
 #endif // _TARGET_*
 
+    if (!IsGcCoverageInterruptInstruction(instrPtr))
```

Comment:
This is the real fix: moving this code here from above the `atCall` setting code.